src/core/lib/channel/channelz_registry.cc

```diff
@@ -131,16 +131,12 @@ char* ChannelzRegistry::InternalGetTopChannels(intptr_t start_channel_id) {
   grpc_json* json = top_level_json;
   grpc_json* json_iterator = nullptr;
   InlinedVector<BaseNode*, 10> top_level_channels;
-  // uuids index into entities one-off (idx 0 is really uuid 1, since 0 is
-  // reserved). However, we want to support requests coming in with
-  // start_channel_id=0, which signifies "give me everything." Hence this
-  // funky looking line below.
-  size_t start_idx = start_channel_id == 0 ? 0 : start_channel_id - 1;
   bool reached_pagination_limit = false;
-  for (size_t i = start_idx; i < entities_.size(); ++i) {
+  for (size_t i = 0; i < entities_.size(); ++i) {
     if (entities_[i] != nullptr &&
         entities_[i]->type() ==
-            grpc_core::channelz::BaseNode::EntityType::kTopLevelChannel) {
+            grpc_core::channelz::BaseNode::EntityType::kTopLevelChannel &&
```

Comment:
I don't quite understand still, where is the binary search?

Reply:
No bin search needed for this step.

The compaction leaves the entities in the vector with increasing uuids, so we just do a linear scan forward and only add after the uuids have passed the start idx.

In the future we could get clever and bin search to shave off some time getting to the first index we check, but no need to prematurely optimize IMO.

Reply:
For a program with 100k channels / servers, you might want it.  When paginating the last results, this will be an O(n^2) operation

Reply:
You mean O(n) from the queriers POV, right? Since they will have to make m requests for however many requests it takes to get the full dataset?

This function's perspective it is O(n), and n will be strictly less than then global_uuid_generator since we do compaction after things unregister themselves

Reply:
I don't feel super strongly about this, but I will note that since we already have the `FindByUuidLocked()` method that implements the binary search, it seems like it should be trivial to use that in these methods.  Is there any reason not to do this?

Reply:
Hmm yes, O(n) iterations for a single query, but O(n^2) to scan the whole list.   I'm also worried that this has to be done under a lock (unless this is an immutable copy).  Under a high number of connections, the size of this list will grow, but more importantly, new channels will be prevented from being added or removed, which means blocking.

Reply:
`FindByUuidLocked ` as is returned -1 if uuid is not present. I will have to modify it to return the first uuid greater or equal to the target.

I can do that now